desafio-07/MatMercer/go/tac.go

```diff
@@ -0,0 +1,132 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+)
+
+func main() {
+	tac()
+}
+
+// in MB size, currently the program uses a max of 2*bufSize iff file size > bufSize
+const maxBufSize = int64(250 << (10 * 2))
+
+var stdout *bufio.Writer
+
+func check(e error) {
+	if e != nil {
+		errAndExit(fmt.Sprintf("tac error: %v", e))
+	}
+}
+
+func errAndExit(msg string) {
+	_, _ = os.Stderr.WriteString(msg + "\n")
+	os.Exit(1)
+}
+
+func tac() {
+	args := os.Args[1:]
+
+	if len(args) != 1 {
+		errAndExit("usage: tac [file]")
+	}
+
+	fileName := args[0]
+
+	f, err := os.Open(fileName)
+	check(err)
+	defer f.Close()
+	fi, err := f.Stat()
+	check(err)
+
+	// buffers stdout by 128kb
+	stdout = bufio.NewWriterSize(os.Stdout, 128<<(10))
+	defer stdout.Flush()
+
+	fs := fi.Size()
+	bufSize := min(fs, maxBufSize)
+	b := make([]byte, bufSize)
```

Comment:
Variáveis com um caracter são uteis em contextos muito restritos. Exemplo: Num for loop ou quando elas saem de contexto em algumas linhas. Aqui, eleas são usadas bem abaixo, o que dificulta a compreensão.